runtime/src/main/cpp/KString.cpp

```diff
@@ -50,9 +50,6 @@ OBJ_GETTER(utf8ToUtf16Impl, const char* rawString, const char* end, uint32_t cha
 template<utf16to8 conversion>
 OBJ_GETTER(utf16ToUtf8Impl, KString thiz, KInt start, KInt size) {
   RuntimeAssert(thiz->type_info() == theStringTypeInfo, "Must use String");
-  if (start < 0 || size < 0 || (size > static_cast<KInt>(thiz->count_) - start)) {
```

Comment:
Boundary checks moved to call site:
 * to be able to throw different exceptions in old and new api in accordance with their contracts
 * to avoid redundant checks for simplified overloads

Reply:
I somewhat dislike this kind of changes, as it allows easily making mistakes if new consumers of API will appear.

Reply:
It's a widespread approach to have an internal implementation without bound checking and check them only on the boundary of public API.
What do you propose instead to achieve this goal?

Reply:
Somehow limit scope of unsafe API and mark its unsafeness in the name.

Reply:
Restored in PR #2973